grpc-server-security-jwt/src/main/java/io/micronaut/grpc/server/security/jwt/GrpcServerSecurityJwtConfiguration.java

```diff
@@ -0,0 +1,106 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.grpc.server.security.jwt;
+
+import edu.umd.cs.findbugs.annotations.Nullable;
+import io.grpc.Status;
+import io.micronaut.context.annotation.ConfigurationProperties;
+import io.micronaut.context.annotation.Requires;
+import io.micronaut.core.bind.annotation.Bindable;
+import io.micronaut.core.order.Ordered;
+import io.micronaut.core.util.StringUtils;
+import io.micronaut.core.util.Toggleable;
+import io.micronaut.grpc.server.GrpcServerConfiguration;
+import io.micronaut.security.config.InterceptUrlMapPattern;
+import io.micronaut.security.config.SecurityConfigurationProperties;
+import io.micronaut.security.token.config.TokenConfigurationProperties;
+import io.micronaut.security.token.jwt.config.JwtConfigurationProperties;
+
+import javax.validation.constraints.NotBlank;
+import java.util.Collection;
+
+/**
+ * gRPC Security JWT configuration.
+ *
+ * @since 2.4.0
+ * @author Brian Wyka
+ */
+@ConfigurationProperties(GrpcServerSecurityJwtConfiguration.PREFIX)
+@Requires(property = SecurityConfigurationProperties.PREFIX + ".enabled", notEquals = StringUtils.FALSE)
+@Requires(property = TokenConfigurationProperties.PREFIX + ".enabled", notEquals = StringUtils.FALSE)
+@Requires(property = JwtConfigurationProperties.PREFIX + ".enabled", notEquals = StringUtils.FALSE)
+@Requires(property = GrpcServerSecurityJwtConfiguration.PREFIX + ".enabled", notEquals = StringUtils.FALSE)
+public interface GrpcServerSecurityJwtConfiguration extends Toggleable {
+
+    /**
+     * The configuration prefix.
+     */
+    String PREFIX = GrpcServerConfiguration.PREFIX + ".security.token.jwt";
+
+    /**
+     * The default name for the JWT metadata key.
+     */
+    String DEFAULT_METADATA_KEY_NAME = "JWT";
+
+    /**
+     * The order to be applied to the server interceptor in the interceptor chain.  Defaults
+     * to {@value io.micronaut.core.order.Ordered#HIGHEST_PRECEDENCE} if not configured.
+     *
+     * @return the order
+     */
+    @Bindable(defaultValue = "" + Ordered.HIGHEST_PRECEDENCE)
+    int getInterceptorOrder();
+
+    /**
+     * Get the list of fully qualified RPC method patterns which should be intercepted and interrogated for a valid JWT.
+     * If no values are provided, by default, all methods will be intercepted.
+     *
+     * @see io.grpc.MethodDescriptor#getFullMethodName()
+     *
+     * @return the intercept method names.
+     */
+    @Nullable
+    Collection<InterceptUrlMapPattern> getInterceptMethodPatterns();
```

Comment:
Why not use the existing configuration?

Reply:
@jameskleeh, don't you think it would be confusing to have HTTP URIs and gRPC "method names" in the same configuration?  I'm open to it, just wanted to limit confusion.

For sample, it might look something like this if you have netty server and grpc server:

```
micronaut.security:
  intercept-url-map:
    - pattern: "/api/resource/**"
      method: GET
      access:
        - isAuthenticated()
    - pattern: "hello.Example/GetWorld"
      access:
        - isAuthenticated()
    - pattern: "/**"
      access:
        - isAnonymous()
```

Also, `HTTP method` does not apply here for gRPC...

Reply:
In your case method is also available to be configured, but given its not used at all for GRPC I think that is fine

Given the variable is called "pattern", I think it still logically applies so I would say use the existing config

Reply:
Sounds good.  I'll make that change.

Reply:
Completed this in https://github.com/micronaut-projects/micronaut-grpc/pull/322/commits/1143ca871680b57604b2e160c0f9f6d10b2cc523